File: 프로그래머스 풀이/lv2_기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int i = 0, size = progresses.size();
    int cal[size];
    
    if ( size == 1 ) {
        answer.push_back(1);
        return answer;
    }
    for ( ; i < size; i++ )
        cal[i] = (99 - progresses[i]) / speeds[i] + 1;
    
    i = 1;
    int current = cal[0];
    answer.push_back(1);

    while ( i < size ) {
        if ( current >= cal[i] )   answer.back()++;
        else {
            current = cal[i];
            answer.push_back(1);
        }
        i++;
    }
    return answer;
}

File: 프로그래머스 풀이/lv2_기능개발.py
import math
def solution(progresses, speeds):
    answer = []
    count, i = 0, 0
    while i < len(progresses) :
        
        days = math.ceil((100 - progresses[i]) / speeds[i])
        count, b = 1, False
        i += 1
        j = i
        for s in speeds[i:] :
            progresses[j] += s*days
            if b == False and progresses[j] >= 100 :
                i += 1
                count += 1
            elif progresses[j] < 100 :
                b = True
            j += 1
        answer.append(count)
        count = 0
    return answer